slatedb/src/bytes_range.rs

```diff
@@ -1,15 +1,27 @@
 use bytes::Bytes;
+use serde::Serialize;
 use std::ops::Bound::{Excluded, Included, Unbounded};
 use std::ops::{Bound, RangeBounds};
 
+use crate::comparable_range::ComparableRange;
+
 /// Concrete struct representing a range of Bytes. Gets around much of
 /// the cumbersome work associated with the generic trait RangeBounds<Bytes>
-#[derive(Debug, Clone, PartialEq)]
+#[derive(Debug, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
 pub(crate) struct BytesRange {
-    start_bound: Bound<Bytes>,
-    end_bound: Bound<Bytes>,
+    inner: ComparableRange<Bytes>,
+}
+
+impl Serialize for BytesRange {
```

Comment:
where are we serializing this type?

Reply:
In admin CLI - https://github.com/slatedb/slatedb/blob/main/slatedb/src/admin.rs#L43

The type is now part of the manifest and we need to be able to convert manifest to json for debug purpose.